dora/tests/src/test/java/alluxio/client/fs/DoraFileSystemIntegrationTest.java

```diff
@@ -0,0 +1,189 @@
+        .standard()
+        .withPathStyleAccessEnabled(true)
+        .withCredentials(
+            new AWSStaticCredentialsProvider(
+                new BasicAWSCredentials(mS3Proxy.getAccessKey(), mS3Proxy.getSecretKey())))
+        .withEndpointConfiguration(
+            new AwsClientBuilder.EndpointConfiguration(mS3Proxy.getUri().toString(),
+                Regions.US_WEST_2.getName()))
+        .build();
+    mS3Client.createBucket(TEST_BUCKET);
+  }
+
+  /**
+   * Writes a file through alluxio into UFS. Deletes the file from UFS.
+   * Read the file with sync interval setting to -1 should give the cached file.
+   * Read the file with sync interval setting to 0 should return error.
+   */
+  @Test
+  public void writeThenDeleteFromUfs() throws IOException, AlluxioException {
+    FileOutStream fos = mFileSystem.createFile(TEST_FILE_URI,
+        CreateFilePOptions.newBuilder().setOverwrite(true).build());
+    fos.write(TEST_CONTENT.getBytes());
+    fos.close();
+
+    mS3Client.deleteObject(TEST_BUCKET, TEST_FILE);
+    assertNotNull(mFileSystem.getStatus(TEST_FILE_URI, GetStatusPOptions.newBuilder()
+        .setCommonOptions(optionNoSync())
+        .build()));
+    try (FileInStream fis = mFileSystem.openFile(TEST_FILE_URI,
+        OpenFilePOptions.newBuilder().setCommonOptions(optionNoSync()).build())) {
+      String content = new String(fis.readAllBytes());
+      assertEquals(TEST_CONTENT, content);
+    }
+
+    assertThrows(FileDoesNotExistException.class, () ->
+        mFileSystem.getStatus(TEST_FILE_URI, GetStatusPOptions.newBuilder()
+            .setCommonOptions(optionSync()).build()));
+
+    assertThrows(FileDoesNotExistException.class, () ->
+        mFileSystem.getStatus(TEST_FILE_URI, GetStatusPOptions.newBuilder()
+            .setCommonOptions(optionNoSync()).build()));
+  }
+
+  /**
+   * Writes a file through alluxio into UFS. Deletes the file from UFS.
+   * Read the file with sync interval setting to -1 should give the cached file.
+   * Read the file with sync interval setting to 0 should return the updated file content.
+   */
+  @Test
+  public void writeThenUpdateFromUfs() throws IOException, AlluxioException {
+    FileOutStream fos = mFileSystem.createFile(TEST_FILE_URI,
+        CreateFilePOptions.newBuilder().setOverwrite(true).build());
+    fos.write(TEST_CONTENT.getBytes());
+    fos.close();
+
+    mS3Client.putObject(TEST_BUCKET, TEST_FILE, UPDATED_TEST_CONTENT);
+    assertNotNull(mFileSystem.getStatus(TEST_FILE_URI, GetStatusPOptions.newBuilder()
+        .setCommonOptions(optionNoSync())
+        .build()));
+
+    try (FileInStream fis = mFileSystem.openFile(TEST_FILE_URI,
+        OpenFilePOptions.newBuilder().setCommonOptions(optionNoSync()).build())) {
+      String content = new String(fis.readAllBytes());
+      assertEquals(TEST_CONTENT, content);
+    }
+
+    mFileSystem.getStatus(TEST_FILE_URI, GetStatusPOptions.newBuilder()
+        .setCommonOptions(optionSync()).build());
```

Comment:
Add comment before and after this statement:
```
// This will update/sync metadata from UFS
  mFileSystem.getStatus(TEST_FILE_URI, GetStatusPOptions.newBuilder()
         .setCommonOptions(optionSync()).build());

// metadata is already updated. Even though we are not going to sync metadata in Read(), it should get the latest content.
```